core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -833,7 +836,8 @@ private[spark] class Executor(
     val message = Heartbeat(executorId, accumUpdates.toArray, env.blockManager.blockManagerId)
     try {
       val response = heartbeatReceiverRef.askSync[HeartbeatResponse](
-          message, RpcTimeout(conf, "spark.executor.heartbeatInterval", "10s"))
+          message, new RpcTimeout(heartbeatInterval.millis.toSeconds.seconds,
+          EXECUTOR_HEARTBEAT_INTERVAL.key))
```

Comment:
Nit: indent, or put the `new RpcTimeout` on this line if it fits. Same comment about using TimeUnit consistently.

Reply:
formatted